### PR TITLE
fix flakiness: reset invalid SpaceRole to valid one in SBR

### PR DESCRIPTION
**Pull request by MatousJobanek**
[SANDBOX-580](https://issues.redhat.com/browse/SANDBOX-580)
https://redhat-internal.slack.com/archives/CHK0J6HT6/p1716912003579899

* reset the SpaceRole to a valid one so it can be removed quickly
* drop the test case with an invalid MUR - we cannot update MUR (webhook blocks that) and it could cause flakiness in the future.

### CODE REVIEW

File: test/e2e/parallel/spacebindingrequest_test.go [hunk 1]
```diff
@@ -123,36 +122,26 @@ func TestCreateSpaceBindingRequest(t *testing.T) {
...
+			_, err = memberAwait.WaitForSpaceBindingRequest(t, client.ObjectKeyFromObject(spaceBindingRequest),
 				UntilSpaceBindingRequestHasConditions(spacebindingrequesttestcommon.UnableToCreateSpaceBinding(fmt.Sprintf("invalid role 'invalid' for space '%s'", space.Name))),
 			)
 			require.NoError(t, err)
 			bindings, err := hostAwait.ListSpaceBindings(space.Name)
 			require.NoError(t, err)
 			assert.Len(t, bindings, 1)
-		})
 
-		t.Run("unable create space binding request with invalid MasterUserRecord", func(t *testing.T) {
```
Comment (mfrancisc) [inline test/e2e/parallel/spacebindingrequest_test.go:135]: I'm not 100% sure I understand why dropping this test case. This is not updating the MUR but creating an SBR with an invalid MUR, which is not covered by the WebHook.

**Review by mfrancisc — COMMENTED** [review]
Thanks for looking into this 🙏 

I'm just not sure we should delete the test case.

**Review by mfrancisc — APPROVED** [review]
Thanks for the fix!

File: test/e2e/parallel/spacebindingrequest_test.go (hunk 1 above)
Comment (MatousJobanek) [reply test/e2e/parallel/spacebindingrequest_test.go:135]: already discussed in https://redhat-internal.slack.com/archives/CHK0J6HT6/p1716974045574919?thread_ts=1716912003.579899&cid=CHK0J6HT6
We cannot reset the MUR back to a valid value (webhook blocks that) so we would end up with an invalid SBR so we could face the same flakiness again. Let's rely on unit-tests in this particular situation

**Pull request merged**